Implement registration

chrisjwelly opened the pull request:
Resolves #1 
Major changes include:
- Unregistered users can now use the sign up button which can be accessed from the sign in page to register.
- Inclusion of the CCA field for sign up and displaying a user's CCA in the index
- Changing the 'login' CSS class into 'pre_session' to be more representative of the code

chrisjwelly committed:
Add sign up button in the sign in page

chrisjwelly committed:
Merge branch 'master' of https://github.com/commit-tech/nussu-logistics into registration

chrisjwelly committed:
Add route to a new view

chrisjwelly committed:
Modify routes and frontend design

chrisjwelly committed:
Use Devise's default functions to display sign up button instead

chrisjwelly committed:
Allow user to enter session after sign up

chrisjwelly committed:
Fix input field to be consistent

chrisjwelly committed:
Remove placeholder view file

chrisjwelly committed:
Make account-wall in sign up be fixed in size

chrisjwelly committed:
Increase height of sign up box slightly

sreycodes commented on app/controllers/registrations_controller.rb:
```diff
@@ -21,4 +22,7 @@ def after_inactive_sign_up_path_for(_resource)
     resource.update_without_password(params)
   end
+  
+  def public_new
```
Do we still need this?

chrisjwelly replied:
No we don't. Thanks for pointing this out! @sreycodes

sreycodes commented on app/views/devise/registrations/new.html.erb:
```diff
@@ -1,57 +1,56 @@
-<div class="container-fluid no-padding">
-  <h1> Add User </h1>
-</div>
+<div class="login">
```
This doesn't seem to be the appropriate class name

chrisjwelly replied:
The view files for ['Forgot your password?'](https://github.com/commit-tech/nussu-logistics/blob/e681fc6d9a9092d07fe0480dcdef9fcec8494db0/app/views/devise/passwords/new.html.erb) (app/views/devise/passwords/new.html.erb) and ['unlock instructions'](https://github.com/commit-tech/nussu-logistics/blob/e681fc6d9a9092d07fe0480dcdef9fcec8494db0/app/views/devise/passwords/new.html.erb) (app/views/devise/unlocks/new.html.erb) use the same class name and I adhered to it for consistency. 

The [login class](https://github.com/commit-tech/nussu-logistics/blob/e681fc6d9a9092d07fe0480dcdef9fcec8494db0/app/assets/stylesheets/login.scss) in the css file also has an extensive definition and I am hesitant to create a new file entirely. An idea I have is instead of calling the class 'login', we can call it 'pre_session'

sreycodes replied:
Sounds like a good idea. Can you port over all the views with this new naming convention?

chrisjwelly replied:
Ported!

sreycodes commented on app/views/devise/registrations/new.html.erb:
```diff
@@ -1,57 +1,56 @@
+
+            <div class="field">
+              <%= f.label :contact_num %>
+              <%= f.text_field :contact_num, :type => "number" %>
```
Should change this to string type

sreycodes committed:
Merge branch 'remove_models' into registration

sreycodes reviewed (changes requested):
<img width="504" alt="image" src="https://user-images.githubusercontent.com/32634781/53295918-87dd4b00-3840-11e9-8a6e-7da870af1e85.png">
1. The "Log In" link seems a bit too cramped.
2. The signup form is not asking for the CCA (field name is "cca" - Check #13)
3. Can you indicate which are the required fields?
4. You need to remove the Add User button from the user list page (/users).

chrisjwelly committed:
Remove obsolete function

chrisjwelly committed:
Remove the 'add user' button which was for admins

chrisjwelly committed:
Merge branch 'update_user_model' of https://github.com/commit-tech/nussu-logistics into registration

chrisjwelly committed:
Add cca field to registration and display cca in index

chrisjwelly committed:
Fix typo

chrisjwelly committed:
Make all fields be required

chrisjwelly committed:
Change contact number to string

chrisjwelly committed:
Modify view to alert about required fields

chrisjwelly committed:
Port over the views with a new naming convention for a css class

chrisjwelly commented:
> <img alt="image" width="504" src="https://user-images.githubusercontent.com/32634781/53295918-87dd4b00-3840-11e9-8a6e-7da870af1e85.png">
> 1. The "Log In" link seems a bit too cramped. 2. The signup form is not asking for the CCA (field name is "cca" - Check #13) 3. Can you indicate which are the required fields? 4. You need to remove the Add User button from the user list page (/users).

@sreycodes 
1. Still trying to resolve this problem. Currently the 'log in' button appears inside the padding and not inside the content so I am a little confused.
2. I have added the CCA field
3. All fields are required now and I have also added a notice regarding this. I made all the fields required because previously only contact number was not required, and I think it makes sense in the context of a booking system to make contact number compulsory. This way, we are able to contact those who are borrowing an item should the need arise.
4. I have removed the add user button

sreycodes committed:
Merge branch 'master' into registration

sreycodes committed:
Merge branch 'registration' of https://github.com/commit-tech/nussu-logistics into registration

sreycodes committed:
Quick fix for login link padding
